python/llm/src/ipex_llm/transformers/pipeline_parallel.py

```diff
@@ -186,6 +218,11 @@ def pipeline_parallel(model, pipeline_parallel_stages):
             model._modules['model'].norm = DummyLayer()
             model._modules['lm_head'] = DummyLayer()
 
+    _enable_lowmem = os.getenv('IPEX_LLM_LOW_MEM')
+    _enable_lowmem = (_enable_lowmem is not None) and (_enable_lowmem.lower() == "1")
+    if _enable_lowmem:
+        model = low_mem_convert(model)
```

Comment:
Not sure this memory optimization could also be added in single GPU inference case? Other LGTM.

Reply:
It could, but for single GPU current existing optimizations may be enough for the demands, so we don't override the CausalLM forward previously.